src/main/kotlin/app/utils/logging/LoggerUtils.kt

```diff
@@ -0,0 +1,157 @@
+package app.utils.logging
+
+/**
+This file had simple json logging utils extending SLF4J in a standard way.
+It brings all our changes into one code module and one test file, and allows us to make a simple override in the
+resources/logback.xml file so that all vagaries are in code and unit-testable; one "just" does this:
+
+<appender name="STDOUT" class="ch.qos.logback.core.ConsoleAppender">
+<encoder class="ch.qos.logback.core.encoder.LayoutWrappingEncoder">
+<layout class="app.utils.logging.LoggerLayoutAppender" />
+</encoder>
+</appender>
+
+See http://logback.qos.ch/manual/layouts.html for examples.
+ */
+
+import ch.qos.logback.classic.spi.ILoggingEvent
+import ch.qos.logback.classic.spi.ThrowableProxyUtil
+import ch.qos.logback.core.CoreConstants
+import ch.qos.logback.core.LayoutBase
+import org.apache.commons.text.StringEscapeUtils
+import org.slf4j.Logger
+import java.net.InetAddress
+import java.text.SimpleDateFormat
+import java.util.*
+
+
+private val defaultFormat = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS") // 2001-07-04T12:08:56.235
+private var topic_name = System.getProperty("topic_name", "NOT_SET")
+private var hostname = InetAddress.getLocalHost().hostName
+private var environment = System.getProperty("environment", "NOT_SET")
+private var application = System.getProperty("application", "NOT_SET")
+private var app_version = System.getProperty("app_version", "NOT_SET")
+private var component = System.getProperty("component", "NOT_SET")
+private var staticData = makeLoggerStaticDataTuples()
+
+fun makeLoggerStaticDataTuples(): String {
+    return "\"topic_name\":\"$topic_name\", " +
+        "\"hostname\":\"$hostname\", " +
+        "\"environment\":\"$environment\", " +
+        "\"application\":\"$application\", " +
+        "\"app_version\":\"$app_version\", " +
+        "\"component\":\"$component\""
+}
+
+fun resetLoggerStaticFieldsForTests() {
+    topic_name = System.getProperty("topic_name", "NOT_SET")
+    hostname = InetAddress.getLocalHost().hostName
+    environment = System.getProperty("environment", "NOT_SET")
+    application = System.getProperty("application", "NOT_SET")
+    app_version = System.getProperty("app_version", "NOT_SET")
+    component = System.getProperty("component", "NOT_SET")
+    staticData = makeLoggerStaticDataTuples()
+}
+
+fun overrideLoggerStaticFieldsForTests(topic: String, host: String, env: String, app: String, version: String, comp: String) {
+    topic_name = topic
+    hostname = host
+    environment = env
+    application = app
+    app_version = version
+    component = comp
+    staticData = makeLoggerStaticDataTuples()
+}
+
+fun logDebug(logger: Logger, message: String, vararg tuples: String) {
```

Comment:
might be worth surrounding the code in this method with logger.isDebugEnabled so that we do't do all the message processing if we're not going to print it

Reply:
Done